pkg/collectsub/client/client.go

```diff
@@ -25,30 +25,36 @@ import (
 	"google.golang.org/grpc/credentials/insecure"
 )
 
-type Client struct {
+type Client interface {
+	AddCollectEntries(ctx context.Context, entries []*pb.CollectEntry) error
+	GetCollectEntries(ctx context.Context, filters []*pb.CollectEntryFilter) ([]*pb.CollectEntry, error)
+	Close()
+}
+
+type client struct {
 	client pb.ColectSubscriberServiceClient
 	conn   *grpc.ClientConn
 }
 
-func NewClient(addr string) (*Client, error) {
+func NewClient(addr string) (Client, error) {
```

Comment:
Is it possible to return a pointer to the `*Client` instead of the `Client` itself?

Reply:
it is an interface now so it is already a pointer